include/civetweb.h

```diff
@@ -334,6 +334,15 @@ mg_set_websocket_handler(struct mg_context *ctx,
                          mg_websocket_close_handler close_handler,
                          void *cbdata);
 
+/* mg_set_auth_handler
+
+   Sets or removes a URI mapping for an authorization handler.
+   This function works similar to mg_set_request_handler - see there. */
+CIVETWEB_API void mg_set_auth_handler(struct mg_context *ctx,
+                                      const char *uri,
+                                      mg_request_handler handler,
```

Comment:
Ok.
The extra handler type will make everything easier to read, and the function names need to be changed to match the new duties of the function.


The most important thing is to get the C interface correctly and do the right check in the right place.
Some fine adjustments can be done by me later as well.


I haven't had a look at C++ yet.